search/include/pcl/search/search.h

```diff
@@ -38,6 +38,7 @@
 
 #pragma once
 
+#include <pcl/pcl_base.h> // for IndicesConstPtr
```

Comment:
Sidenote: should we move pointer-to-indices types to `types.h`? Or we want to avoid including `memory.h` there?

Reply:
> Or we want to avoid including memory.h there?

Mostly this